{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "22a1892d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bd53ce54",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Email de contacto</th>\n",
       "      <th>Fecha de envío</th>\n",
       "      <th>Fecha de recepción</th>\n",
       "      <th>Fecha de emisión de resultados</th>\n",
       "      <th>Autores del análisis</th>\n",
       "      <th>Autor remitente</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>fegarcia@ugr.es</td>\n",
       "      <td>2021-12-21</td>\n",
       "      <td>2021-12-27</td>\n",
       "      <td>2022-01-21</td>\n",
       "      <td>Federico García, Javier Pérez Florido, Adolfo ...</td>\n",
       "      <td>Federico García</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ID Email de contacto Fecha de envío Fecha de recepción  \\\n",
       "0  COD_2129   fegarcia@ugr.es     2021-12-21         2021-12-27   \n",
       "\n",
       "  Fecha de emisión de resultados  \\\n",
       "0                     2022-01-21   \n",
       "\n",
       "                                Autores del análisis  Autor remitente  \n",
       "0  Federico García, Javier Pérez Florido, Adolfo ...  Federico García  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_excel('CONTROL CALIDAD-HERA COD_2129.xlsx',sheet_name=\"2. General\", index_col=None, header=1)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ac5213e7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>ID Muestra</th>\n",
       "      <th>Protocolo diagnóstico SARS-CoV-2</th>\n",
       "      <th>Valor Ct PCR</th>\n",
       "      <th>Variante</th>\n",
       "      <th>Asignación de Linaje</th>\n",
       "      <th>Versión Pango</th>\n",
       "      <th>Read count</th>\n",
       "      <th>%QC filtered</th>\n",
       "      <th>%Reads host</th>\n",
       "      <th>%Reads virus</th>\n",
       "      <th>%Unmapped</th>\n",
       "      <th>% Genome &gt; 10x</th>\n",
       "      <th>Mean depth of coverage value</th>\n",
       "      <th>%Ns</th>\n",
       "      <th>Number of variants (AF &gt; 75%)</th>\n",
       "      <th>Numer of variants with effect (missense, frameshit, stop codon)</th>\n",
       "      <th>Reference genome accession</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COD_2101</td>\n",
       "      <td>#1</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>25</td>\n",
       "      <td>Alpha</td>\n",
       "      <td>B.1.1.7</td>\n",
       "      <td>Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...</td>\n",
       "      <td>393636.0</td>\n",
       "      <td>0.8816</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.97</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.990402</td>\n",
       "      <td>690.0</td>\n",
       "      <td>0.724</td>\n",
       "      <td>42.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>COD_2101</td>\n",
       "      <td>#2</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>23</td>\n",
       "      <td>Beta</td>\n",
       "      <td>B.1.351</td>\n",
       "      <td>Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...</td>\n",
       "      <td>427164.0</td>\n",
       "      <td>0.8932</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.98</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.996689</td>\n",
       "      <td>740.0</td>\n",
       "      <td>0.097</td>\n",
       "      <td>28.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COD_2101</td>\n",
       "      <td>#3</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>29</td>\n",
       "      <td>19B</td>\n",
       "      <td>A.28</td>\n",
       "      <td>Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...</td>\n",
       "      <td>248256.0</td>\n",
       "      <td>0.8962</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.98</td>\n",
       "      <td>1.95</td>\n",
       "      <td>0.982811</td>\n",
       "      <td>480.0</td>\n",
       "      <td>1.400</td>\n",
       "      <td>24.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>COD_2101</td>\n",
       "      <td>#4</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>25</td>\n",
       "      <td>Mu</td>\n",
       "      <td>B.1.621</td>\n",
       "      <td>Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...</td>\n",
       "      <td>348634.0</td>\n",
       "      <td>0.8958</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.98</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.990101</td>\n",
       "      <td>635.0</td>\n",
       "      <td>0.668</td>\n",
       "      <td>38.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>COD_2101</td>\n",
       "      <td>#5</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>27</td>\n",
       "      <td>Gamma</td>\n",
       "      <td>P.1</td>\n",
       "      <td>Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...</td>\n",
       "      <td>418158.0</td>\n",
       "      <td>0.8998</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.99</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.990804</td>\n",
       "      <td>735.0</td>\n",
       "      <td>0.690</td>\n",
       "      <td>40.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>COD_2109</td>\n",
       "      <td>#6</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>31</td>\n",
       "      <td>Delta</td>\n",
       "      <td>AY.9.2</td>\n",
       "      <td>Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...</td>\n",
       "      <td>380354.0</td>\n",
       "      <td>0.8960</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.99</td>\n",
       "      <td>1.95</td>\n",
       "      <td>0.980236</td>\n",
       "      <td>637.0</td>\n",
       "      <td>1.642</td>\n",
       "      <td>36.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>COD_2109</td>\n",
       "      <td>#7</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>36</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>270042.0</td>\n",
       "      <td>0.8657</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.96</td>\n",
       "      <td>90.78</td>\n",
       "      <td>0.092165</td>\n",
       "      <td>182.0</td>\n",
       "      <td>90.330</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>COD_2109</td>\n",
       "      <td>#8</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>21</td>\n",
       "      <td>Delta</td>\n",
       "      <td>AY.94</td>\n",
       "      <td>Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...</td>\n",
       "      <td>223896.0</td>\n",
       "      <td>0.8960</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.98</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.990135</td>\n",
       "      <td>434.0</td>\n",
       "      <td>0.681</td>\n",
       "      <td>45.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>COD_2109</td>\n",
       "      <td>#9</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>23</td>\n",
       "      <td>Delta</td>\n",
       "      <td>AY.94</td>\n",
       "      <td>Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...</td>\n",
       "      <td>339808.0</td>\n",
       "      <td>0.8858</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.98</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.990135</td>\n",
       "      <td>611.0</td>\n",
       "      <td>0.748</td>\n",
       "      <td>45.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>COD_2109</td>\n",
       "      <td>#10</td>\n",
       "      <td>Whole genome</td>\n",
       "      <td>29</td>\n",
       "      <td>Delta</td>\n",
       "      <td>AY.43</td>\n",
       "      <td>Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...</td>\n",
       "      <td>315836.0</td>\n",
       "      <td>0.8514</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.98</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.989867</td>\n",
       "      <td>563.0</td>\n",
       "      <td>0.741</td>\n",
       "      <td>43.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>MN908947.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gen N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ID ID Muestra Protocolo diagnóstico SARS-CoV-2 Valor Ct PCR  \\\n",
       "0   COD_2101         #1                     Whole genome           25   \n",
       "1   COD_2101         #2                     Whole genome           23   \n",
       "2   COD_2101         #3                     Whole genome           29   \n",
       "3   COD_2101         #4                     Whole genome           25   \n",
       "4   COD_2101         #5                     Whole genome           27   \n",
       "5   COD_2109         #6                     Whole genome           31   \n",
       "6   COD_2109         #7                     Whole genome           36   \n",
       "7   COD_2109         #8                     Whole genome           21   \n",
       "8   COD_2109         #9                     Whole genome           23   \n",
       "9   COD_2109        #10                     Whole genome           29   \n",
       "10       NaN        NaN                              NaN        Gen N   \n",
       "\n",
       "   Variante Asignación de Linaje  \\\n",
       "0     Alpha              B.1.1.7   \n",
       "1      Beta              B.1.351   \n",
       "2       19B                 A.28   \n",
       "3        Mu              B.1.621   \n",
       "4     Gamma                  P.1   \n",
       "5     Delta               AY.9.2   \n",
       "6         -                    -   \n",
       "7     Delta                AY.94   \n",
       "8     Delta                AY.94   \n",
       "9     Delta                AY.43   \n",
       "10      NaN                  NaN   \n",
       "\n",
       "                                        Versión Pango  Read count  \\\n",
       "0   Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...    393636.0   \n",
       "1   Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...    427164.0   \n",
       "2   Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...    248256.0   \n",
       "3   Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...    348634.0   \n",
       "4   Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...    418158.0   \n",
       "5   Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...    380354.0   \n",
       "6                                                   -    270042.0   \n",
       "7   Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...    223896.0   \n",
       "8   Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...    339808.0   \n",
       "9   Pangolin 3.1.17 / PangoLEARN 2021-12-06 / pang...    315836.0   \n",
       "10                                                NaN         NaN   \n",
       "\n",
       "    %QC filtered  %Reads host  %Reads virus  %Unmapped  % Genome > 10x  \\\n",
       "0         0.8816          NaN         99.97       0.95        0.990402   \n",
       "1         0.8932          NaN         99.98       0.31        0.996689   \n",
       "2         0.8962          NaN         99.98       1.95        0.982811   \n",
       "3         0.8958          NaN         99.98       0.96        0.990101   \n",
       "4         0.8998          NaN         99.99       0.93        0.990804   \n",
       "5         0.8960          NaN         99.99       1.95        0.980236   \n",
       "6         0.8657          NaN         99.96      90.78        0.092165   \n",
       "7         0.8960          NaN         99.98       0.97        0.990135   \n",
       "8         0.8858          NaN         99.98       0.96        0.990135   \n",
       "9         0.8514          NaN         99.98       0.95        0.989867   \n",
       "10           NaN          NaN           NaN        NaN             NaN   \n",
       "\n",
       "    Mean depth of coverage value     %Ns  Number of variants (AF > 75%)  \\\n",
       "0                          690.0   0.724                           42.0   \n",
       "1                          740.0   0.097                           28.0   \n",
       "2                          480.0   1.400                           24.0   \n",
       "3                          635.0   0.668                           38.0   \n",
       "4                          735.0   0.690                           40.0   \n",
       "5                          637.0   1.642                           36.0   \n",
       "6                          182.0  90.330                            1.0   \n",
       "7                          434.0   0.681                           45.0   \n",
       "8                          611.0   0.748                           45.0   \n",
       "9                          563.0   0.741                           43.0   \n",
       "10                           NaN     NaN                            NaN   \n",
       "\n",
       "    Numer of variants with effect (missense, frameshit, stop codon)  \\\n",
       "0                                                30.0                 \n",
       "1                                                20.0                 \n",
       "2                                                13.0                 \n",
       "3                                                22.0                 \n",
       "4                                                27.0                 \n",
       "5                                                24.0                 \n",
       "6                                                 1.0                 \n",
       "7                                                30.0                 \n",
       "8                                                30.0                 \n",
       "9                                                30.0                 \n",
       "10                                                NaN                 \n",
       "\n",
       "   Reference genome accession  \n",
       "0                  MN908947.3  \n",
       "1                  MN908947.3  \n",
       "2                  MN908947.3  \n",
       "3                  MN908947.3  \n",
       "4                  MN908947.3  \n",
       "5                  MN908947.3  \n",
       "6                  MN908947.3  \n",
       "7                  MN908947.3  \n",
       "8                  MN908947.3  \n",
       "9                  MN908947.3  \n",
       "10                        NaN  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_excel('CONTROL CALIDAD-HERA COD_2129.xlsx',sheet_name=\"3. Resultados\", index_col=None, header=2)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9a244a0d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>LAB ID</th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Sequencing date</th>\n",
       "      <th>Enrichment protocol</th>\n",
       "      <th>if enrichment protocol is other, specify</th>\n",
       "      <th>Amplicon protocol</th>\n",
       "      <th>if amplicon protocol if other, especify</th>\n",
       "      <th>Amplicon version</th>\n",
       "      <th>Library preparation kit</th>\n",
       "      <th>if library prepation is other, specify</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 1014</th>\n",
       "      <th>Unnamed: 1015</th>\n",
       "      <th>Unnamed: 1016</th>\n",
       "      <th>Unnamed: 1017</th>\n",
       "      <th>Unnamed: 1018</th>\n",
       "      <th>Unnamed: 1019</th>\n",
       "      <th>Unnamed: 1020</th>\n",
       "      <th>Unnamed: 1021</th>\n",
       "      <th>Unnamed: 1022</th>\n",
       "      <th>Unnamed: 1023</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#1</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#2</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#3</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#4</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#5</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#6</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#7</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#8</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#9</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>COD_2101_#10</td>\n",
       "      <td>2021-12-30</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARTIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>Illumina DNA Prep Tagmentation</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 1024 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     LAB ID     Sample ID Sequencing date Enrichment protocol  \\\n",
       "0  COD_2129   COD_2101_#1      2021-12-30            Amplicon   \n",
       "1  COD_2129   COD_2101_#2      2021-12-30            Amplicon   \n",
       "2  COD_2129   COD_2101_#3      2021-12-30            Amplicon   \n",
       "3  COD_2129   COD_2101_#4      2021-12-30            Amplicon   \n",
       "4  COD_2129   COD_2101_#5      2021-12-30            Amplicon   \n",
       "5  COD_2129   COD_2101_#6      2021-12-30            Amplicon   \n",
       "6  COD_2129   COD_2101_#7      2021-12-30            Amplicon   \n",
       "7  COD_2129   COD_2101_#8      2021-12-30            Amplicon   \n",
       "8  COD_2129   COD_2101_#9      2021-12-30            Amplicon   \n",
       "9  COD_2129  COD_2101_#10      2021-12-30            Amplicon   \n",
       "\n",
       "   if enrichment protocol is other, specify Amplicon protocol  \\\n",
       "0                                       NaN             ARTIC   \n",
       "1                                       NaN             ARTIC   \n",
       "2                                       NaN             ARTIC   \n",
       "3                                       NaN             ARTIC   \n",
       "4                                       NaN             ARTIC   \n",
       "5                                       NaN             ARTIC   \n",
       "6                                       NaN             ARTIC   \n",
       "7                                       NaN             ARTIC   \n",
       "8                                       NaN             ARTIC   \n",
       "9                                       NaN             ARTIC   \n",
       "\n",
       "   if amplicon protocol if other, especify  Amplicon version  \\\n",
       "0                                      NaN                 3   \n",
       "1                                      NaN                 3   \n",
       "2                                      NaN                 3   \n",
       "3                                      NaN                 3   \n",
       "4                                      NaN                 3   \n",
       "5                                      NaN                 3   \n",
       "6                                      NaN                 3   \n",
       "7                                      NaN                 3   \n",
       "8                                      NaN                 3   \n",
       "9                                      NaN                 3   \n",
       "\n",
       "          Library preparation kit  if library prepation is other, specify  \\\n",
       "0  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "1  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "2  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "3  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "4  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "5  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "6  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "7  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "8  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "9  Illumina DNA Prep Tagmentation                                     NaN   \n",
       "\n",
       "   ... Unnamed: 1014 Unnamed: 1015  Unnamed: 1016  Unnamed: 1017  \\\n",
       "0  ...           NaN           NaN            NaN            NaN   \n",
       "1  ...           NaN           NaN            NaN            NaN   \n",
       "2  ...           NaN           NaN            NaN            NaN   \n",
       "3  ...           NaN           NaN            NaN            NaN   \n",
       "4  ...           NaN           NaN            NaN            NaN   \n",
       "5  ...           NaN           NaN            NaN            NaN   \n",
       "6  ...           NaN           NaN            NaN            NaN   \n",
       "7  ...           NaN           NaN            NaN            NaN   \n",
       "8  ...           NaN           NaN            NaN            NaN   \n",
       "9  ...           NaN           NaN            NaN            NaN   \n",
       "\n",
       "  Unnamed: 1018  Unnamed: 1019  Unnamed: 1020  Unnamed: 1021  Unnamed: 1022  \\\n",
       "0           NaN            NaN            NaN            NaN            NaN   \n",
       "1           NaN            NaN            NaN            NaN            NaN   \n",
       "2           NaN            NaN            NaN            NaN            NaN   \n",
       "3           NaN            NaN            NaN            NaN            NaN   \n",
       "4           NaN            NaN            NaN            NaN            NaN   \n",
       "5           NaN            NaN            NaN            NaN            NaN   \n",
       "6           NaN            NaN            NaN            NaN            NaN   \n",
       "7           NaN            NaN            NaN            NaN            NaN   \n",
       "8           NaN            NaN            NaN            NaN            NaN   \n",
       "9           NaN            NaN            NaN            NaN            NaN   \n",
       "\n",
       "   Unnamed: 1023  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  \n",
       "3            NaN  \n",
       "4            NaN  \n",
       "5            NaN  \n",
       "6            NaN  \n",
       "7            NaN  \n",
       "8            NaN  \n",
       "9            NaN  \n",
       "\n",
       "[10 rows x 1024 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_excel('CONTROL CALIDAD-HERA COD_2129.xlsx',sheet_name=\"4. Datos de secuenciación \", index_col=None, header=1)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "80c68ebd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Bioinformatic protocol</th>\n",
       "      <th>if bioinformatic protocol is other, specify</th>\n",
       "      <th>Bioinformatic protocol version</th>\n",
       "      <th>Commercial/open-source/both</th>\n",
       "      <th>Preprocessing</th>\n",
       "      <th>if preprocessing other</th>\n",
       "      <th>Preprocessing params</th>\n",
       "      <th>Mapping</th>\n",
       "      <th>...</th>\n",
       "      <th>Assembly params</th>\n",
       "      <th>Variant Calling</th>\n",
       "      <th>if variant calling other</th>\n",
       "      <th>Variant Calling params</th>\n",
       "      <th>Consensus</th>\n",
       "      <th>if consensus other</th>\n",
       "      <th>Consensus criteria</th>\n",
       "      <th>Linage identification</th>\n",
       "      <th>if lineage identification other</th>\n",
       "      <th>Quality control metrics (sample discard criteria)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#1</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#2</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#3</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#4</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#5</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#6</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#7</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#8</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#9</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>COD_2129</td>\n",
       "      <td>#10</td>\n",
       "      <td>nf-core/viralrecon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.1.0</td>\n",
       "      <td>open-source</td>\n",
       "      <td>fastp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--detect_adapter_for_pe --cut_front --cut_tail...</td>\n",
       "      <td>bowtie 2.4.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ivar 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10</td>\n",
       "      <td>ivar consensus 1.2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-q 20 -t 25 -m 10 -n N</td>\n",
       "      <td>Pangolin 3.1.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Reference genome covered &lt; 95%</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ID Sample ID Bioinformatic protocol  \\\n",
       "0  COD_2129        #1     nf-core/viralrecon   \n",
       "1  COD_2129        #2     nf-core/viralrecon   \n",
       "2  COD_2129        #3     nf-core/viralrecon   \n",
       "3  COD_2129        #4     nf-core/viralrecon   \n",
       "4  COD_2129        #5     nf-core/viralrecon   \n",
       "5  COD_2129        #6     nf-core/viralrecon   \n",
       "6  COD_2129        #7     nf-core/viralrecon   \n",
       "7  COD_2129        #8     nf-core/viralrecon   \n",
       "8  COD_2129        #9     nf-core/viralrecon   \n",
       "9  COD_2129       #10     nf-core/viralrecon   \n",
       "\n",
       "   if bioinformatic protocol is other, specify Bioinformatic protocol version  \\\n",
       "0                                          NaN                          1.1.0   \n",
       "1                                          NaN                          1.1.0   \n",
       "2                                          NaN                          1.1.0   \n",
       "3                                          NaN                          1.1.0   \n",
       "4                                          NaN                          1.1.0   \n",
       "5                                          NaN                          1.1.0   \n",
       "6                                          NaN                          1.1.0   \n",
       "7                                          NaN                          1.1.0   \n",
       "8                                          NaN                          1.1.0   \n",
       "9                                          NaN                          1.1.0   \n",
       "\n",
       "  Commercial/open-source/both Preprocessing  if preprocessing other  \\\n",
       "0                 open-source         fastp                     NaN   \n",
       "1                 open-source         fastp                     NaN   \n",
       "2                 open-source         fastp                     NaN   \n",
       "3                 open-source         fastp                     NaN   \n",
       "4                 open-source         fastp                     NaN   \n",
       "5                 open-source         fastp                     NaN   \n",
       "6                 open-source         fastp                     NaN   \n",
       "7                 open-source         fastp                     NaN   \n",
       "8                 open-source         fastp                     NaN   \n",
       "9                 open-source         fastp                     NaN   \n",
       "\n",
       "                                Preprocessing params       Mapping  ...  \\\n",
       "0  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "1  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "2  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "3  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "4  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "5  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "6  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "7  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "8  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "9  --detect_adapter_for_pe --cut_front --cut_tail...  bowtie 2.4.1  ...   \n",
       "\n",
       "   Assembly params Variant Calling if variant calling other  \\\n",
       "0              NaN      ivar 1.2.2                      NaN   \n",
       "1              NaN      ivar 1.2.2                      NaN   \n",
       "2              NaN      ivar 1.2.2                      NaN   \n",
       "3              NaN      ivar 1.2.2                      NaN   \n",
       "4              NaN      ivar 1.2.2                      NaN   \n",
       "5              NaN      ivar 1.2.2                      NaN   \n",
       "6              NaN      ivar 1.2.2                      NaN   \n",
       "7              NaN      ivar 1.2.2                      NaN   \n",
       "8              NaN      ivar 1.2.2                      NaN   \n",
       "9              NaN      ivar 1.2.2                      NaN   \n",
       "\n",
       "   Variant Calling params             Consensus if consensus other  \\\n",
       "0       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "1       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "2       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "3       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "4       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "5       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "6       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "7       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "8       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "9       -q 20 -t 25 -m 10  ivar consensus 1.2.2                NaN   \n",
       "\n",
       "       Consensus criteria Linage identification  \\\n",
       "0  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "1  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "2  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "3  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "4  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "5  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "6  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "7  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "8  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "9  -q 20 -t 25 -m 10 -n N       Pangolin 3.1.17   \n",
       "\n",
       "  if lineage identification other  \\\n",
       "0                             NaN   \n",
       "1                             NaN   \n",
       "2                             NaN   \n",
       "3                             NaN   \n",
       "4                             NaN   \n",
       "5                             NaN   \n",
       "6                             NaN   \n",
       "7                             NaN   \n",
       "8                             NaN   \n",
       "9                             NaN   \n",
       "\n",
       "   Quality control metrics (sample discard criteria)  \n",
       "0                     Reference genome covered < 95%  \n",
       "1                     Reference genome covered < 95%  \n",
       "2                     Reference genome covered < 95%  \n",
       "3                     Reference genome covered < 95%  \n",
       "4                     Reference genome covered < 95%  \n",
       "5                     Reference genome covered < 95%  \n",
       "6                     Reference genome covered < 95%  \n",
       "7                     Reference genome covered < 95%  \n",
       "8                     Reference genome covered < 95%  \n",
       "9                     Reference genome covered < 95%  \n",
       "\n",
       "[10 rows x 24 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_excel('CONTROL CALIDAD-HERA COD_2129.xlsx',sheet_name=\"5. Datos  Bioinformaticos\", index_col=None, header=1)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71a4f8d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
